.github/workflows/push.yml

```diff
@@ -0,0 +1,59 @@
+name: Push workflow
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  push:
+    name: openrate-consumer
+    runs-on: ubuntu-latest
+    steps:
+      - name: Read version from version file
+        id: read_version
+        uses: christian-draeger/read-properties@1.0.1
+        with:
+          path: 'version.properties'
+          property: 'version'
+
+      - name: Bump version
+        id: bump_version
+        uses: christian-draeger/increment-semantic-version@1.0.1
+        with:
+          current-version: ${{ steps.read_version.outputs.value }}
+          version-fragment: 'bug'
```

Comment:
If I read this correctly, a merge to master will always do a bug/patch version bump. How do we do a minor/major version release?

Reply:
Hmm yes, this one will only bump the patch version. I am using this action to ensure that we are able to push the image before we commit the version bump, so we dont end up with skipped versions. As for how we can do a minor/major version release, an idea I have is that we can instead have it use the "old" version to push an image(basically the version in the version.properties before the workflow runs), then the workflow will bump the version, and that version will be used for the next time the docker image is built. This would allow us to bump the major/minor version manually manually without skipping a version. (so when we bump it to version 2.0.0 manually, the image will be tagged with 2.0.0, and not 2.0.1, and 2.0.1 will be used the next time)

Reply:
There are probably other solutions. This action(https://github.com/phips28/gh-action-bump-version) is being used in other projects of ours, and does a patch/minor/major bump by detecting certain keywords in a commit message, but it does not allow us to separate bumping the version from comitting it to the repository, which leads to possible skipped versions if we fail to build/push the docker image for some reason.

Reply:
One thing we can also do, is to simply have two branches, "dev" and "prod" for example, and always use the latest version. We do PRs from feature branches to the dev branch, and when we want to put something in production after testing it, we can do PRs from the dev branch to the prod branch. This way we dont need to worry about versioning.

Reply:
I think it is better to approve this PR now, and then make a JIRA issue for this versioning dilemma, to be looked into later. I would like to get probe up and running

Reply:
Ok, we can merge this as is for now I think, but would like to do a discussion on how to handle versioning. Could you open a ticket in Jira?

One example is how we do it for https://github.oslo.kommune.no/origo-dataplatform/common-python/blob/master/Jenkinsfile - The user does a manual release on the master branch, and Jenkins checks if the last commit is a version bump and if so builds and pushes that to Nexus.

Reply:
How about using the git hash as part of the version number for these automagic releases, then having a different process for the manually curated releases?